src/DiscGolfScorecard1.java

```diff
@@ -0,0 +1,365 @@
+ *             $this.rep have entries >= 0]
+ * @correspondence this = [holes in this.rep with initial starting hole
+ *                 $this.startingHoleNumber and current hole
+ *                 $this.currentHoleNumber]
+ *
+ * @author Luke Zhang
+ */
+public class DiscGolfScorecard1 extends DiscGolfScorecardSecondary {
+
+    /*
+     * Private members ---------------------------------------------------------
+     */
+
+    /**
+     * Default length of scorecard.
+     */
+    private static final int DEFAULT_SCORECARD_LENGTH = 18;
+
+    /**
+     * Default starting hole.
+     */
+    private static final int DEFAULT_STARTING_HOLE = 1;
+
+    /**
+     * Default par.
+     */
+    private static final int DEFAULT_PAR = 3;
+
+    /**
+     * Default distance.
+     */
+    private static final int DEFAULT_DISTANCE = 0;
+
+    /**
+     * Default strokes.
+     */
+    private static final int DEFAULT_STROKES = 0;
+
+    /**
+     * Current hole number.
+     */
+    private int currentHoleNumber;
+
+    /**
+     * Starting hole number.
+     */
+    private int startingHoleNumber;
+
+    /**
+     * Representation of {@code this}.
+     */
+    private Hole[] rep;
+
+    /**
+     * Creator of inital representation.
+     *
+     * @param startingHole
+     *            the first hole returned in scorecard
+     * @param distances
+     *            array of the distances for each hole
+     * @param pars
+     *            array of the pars for each hole
+     * @param strokes
+     *            array of the strokes for each hole
+     * @requires |pars| = |distances| = |strokes| and |pars| > 0 and 0 &lt;
+     *           startingHole <= |pars| and no entries in pars or distances or
+     *           strokes can be < 0
+     * @ensures |this.rep| = |pars| and this.scorecard[is a scorecard with holes
+     *          with the specified pars, distances, and strokes]
+     */
+    private void createNewRep(int startingHole, int[] pars, int[] distances,
+            int[] strokes) {
+        assert pars.length == distances.length
+                && distances.length == strokes.length : "Violation of "
+                        + "[|pars| = |distances| = |strokes|]";
+        assert pars.length > 0 : "Violation of [|pars| > 0]";
+        assert 0 < startingHole && startingHole <= pars.length : "Violation of "
+                + "[0 < startingHole <= |pars|]";
+        //Precondition of no entries < 0 in pars, distances or strokes not checked
+        this.rep = new Hole[pars.length];
+        this.startingHoleNumber = startingHole;
+        this.currentHoleNumber = startingHole;
+        for (int i = 0; i < pars.length; i++) {
+            this.rep[i] = new Hole(pars[i], distances[i], strokes[i]);
+        }
+    }
+
+    /*
+     * Constructors ------------------------------------------------------------
+     */
+
+    /**
+     * No argument constructor.
+     */
+    public DiscGolfScorecard1() {
```

Comment:
I would put most of this stuff in createNewRep. Though, I don't think it matters either way.